src/amo/sagas/reviews.js

```diff
@@ -281,21 +281,23 @@ function* manageAddonReview(
       yield put(hideEditReviewForm({ reviewId: reviewFromResponse.id }));
     }
 
-    invariant(reviewFromResponse.version, 'version is required');
-    yield put(
-      setLatestReview({
-        addonId: reviewFromResponse.addon.id,
-        addonSlug: reviewFromResponse.addon.slug,
-        review: reviewFromResponse,
-        userId: reviewFromResponse.user.id,
-        versionId: reviewFromResponse.version.id,
-      }),
-    );
+    if (!reviewFromResponse.is_developer_reply) {
```

Comment:
I did this because `reviewFromResponse.version` is missing and therefore the `invariant` is being triggered. I realized that if we are updating a reply we really don't need to do either of these things (`setLatestReview` or `fetchAddon`), so it made sense to just not do them.

Looking closer now, I'm not sure that the fact that `version` is missing from the response is expected, and maybe that's a bug on the server side. The [docs](https://addons-server.readthedocs.io/en/latest/topics/api/ratings.html#rating-detail-object) don't mention that `version` is optional. I guess I'll file a bug on addons-server for that, but it still makes sense to me that we wouldn't dispatch either of those actions for a reply update as neither are needed.

@kumar303 What do you think?

Reply:
I opened https://github.com/mozilla/addons-server/issues/9550 to report the missing `version`, although I now suspect that that is expected.

Reply:
Interesting. It looks like someone had already figured out that the version could be missing because our Flow type is correct:

https://github.com/mozilla/addons-frontend/blob/a238ae43335c366c34569a29d6c021130ef982c0/src/amo/api/reviews.js#L31-L43